src/main/java/com/moujounakki/archimedesfluids/FluidPool.java

```diff
@@ -7,126 +7,178 @@
 import net.minecraft.world.level.material.Fluid;
 import net.minecraft.world.level.material.FluidState;
 
+import java.util.ArrayDeque;
 import java.util.HashSet;
-import java.util.LinkedList;
+import java.util.Queue;
+import java.util.Set;
 
 public class FluidPool {
     private final Level level;
     private final Fluid fluid;
-    private final HashSet<BlockPos> explored = new HashSet<>();
-    private final LinkedList<BlockPos> unexplored = new LinkedList<>();
-    private final HashSet<BlockPos> banned = new HashSet<>();
+    private final Set<BlockPos> explored = new HashSet<>();
+    private final Queue<BlockPos> unexplored = new ArrayDeque<>();
+    private final Set<BlockPos> banned = new HashSet<>();
```

Comment:
Why was the type replaced with `Set`?

Reply:
That was to make future changes easier if need be, more abstraction etc etc

Reply:
Though again, that one is much less important and moreso for flexibility for future changes without having to redo everything